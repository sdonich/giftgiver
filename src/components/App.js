import React from 'react';
import { Button } from 'react-bootstrap';

class App extends React.Component {
  state = {
    gifts: []
  }

  addGift = () => {
    const { gifts } = this.state;
    const ids = this.state.gifts.map(gift => gift.id);
    const max_id = ids.length > 0 ? Math.max(...ids) : 0;

    this.setState({ gifts: [...gifts, { id: max_id + 1 }] });
  }

  render() {
    return (
      <div>
        <h2>Gift Giver</h2>
        <div className="gift-list">
          {
            this.state.gifts.map(gift => {
              return (
                <div key={gift.id}></div>
              );
            })
          }
        </div>
        <Button className="btn-add" onClick={this.addGift}>Add Gift</Button>
      </div>
    );
  }
}

export default App;

