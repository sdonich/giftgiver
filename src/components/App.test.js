import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import App from './App';

describe('App', () => {
  const app = shallow(<App />);

  test('renders correctly', () => {
    const tree = renderer
      .create(<App />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('initializes the `state` with an empty list of gifts', () => {
    expect(app.state().gifts).toEqual([]);
  });

  describe('when clicking the `add-gift` button', () => {
    beforeEach(() => {
      app.find('.btn-add').simulate('click');
    });

    afterEach(() => {
      app.setState({gifts: []});
    });

    test('adds a new gift to `state`', () => {
      expect(app.state().gifts).toEqual([{ id: 1 }]);
    });

    test('adds a new gift to the rendered list', () => {
      expect(app.find('.gift-list').children().length).toBe(1);
    });

    test('creates a Gift component', () => {
      expect(app.find('Gift').exists()).toBe(true);
    });
  });
});

