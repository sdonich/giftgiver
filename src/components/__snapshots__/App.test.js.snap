// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div>
  <h2>
    Gift Giver
  </h2>
  <div
    className="gift-list"
  />
  <button
    className="btn-add btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Add Gift
  </button>
</div>
`;
